api/src/main/java/jakarta/servlet/http/Cookie.java

```diff
@@ -422,4 +423,101 @@ public void setHttpOnly(boolean isHttpOnly) {
     public boolean isHttpOnly() {
         return isHttpOnly;
     }
+
+    /**
+     * Sets the <i>SameSite</i> attribute for this cookie with the provided value.
+     *
+     * <p>
+     * When <code>sameSite</code> is not <code>null</code>, this cookie will have its <code>SameSite</code>
+     * attribute set with the provided value.
+     *
+     * @param sameSite the <i>SameSite</i> attribute's value; may be null if the attribute should not be set
+     *
+     * @since Servlet 5.0
+     */
+    public void setSameSite(SameSite sameSite) {
+        this.sameSite = sameSite;
+    }
+
+    /**
+     * Returns the <i>SameSite</i> attribute associated with this cookie.
+     *
+     * @return the <i>SameSite</i> attribute associated with this cookie; may be null to indicate that the attribute
+     *         should not be set
+     *
+     * @since Servlet 5.0
+     */
+    public SameSite getSameSite() {
+        return sameSite;
+    }
+
+    /**
+     * The available values that can be used with the cookie's <i>SameSite</i> attribute.
+     */
+    public enum SameSite {
+
+        /**
+         * This mode effectively disables the protection provided by SameSite cookies.
+         *
+         * <p>
+         * When SameSiteMode is {@literal None}, applications should have additional measures to protect cookies
+         * from CSRF attacks.
+         *
+         * <p>
+         * Browsers may default to {@link SameSite#LAX} or {@link SameSite#STRICT} when the flag is not
+         * explicitly set by applications in the Set-Cookie header. Setting the {@literal SameSite} attribute to
+         * {@literal None} ensures that this default behavior is disabled in browsers.
+         *
+         * <p>
+         * Note that this mode is browser dependent, and browsers may require cookies to have the {@literal Secure}
+         * attribute to be set at the same time.
```

Comment:
/*
I wonder if this should mention the horrors of:
https://www.chromium.org/updates/same-site/incompatible-clients
Chances are it would turn outdated eventually though.
*/

Reply:
I think a general warning that browsers may interpret spec differently is fine.... Perhaps we should say that on every feature :)